lib/fs.js

```diff
@@ -1616,7 +1615,7 @@ fs.mkdtemp = function(prefix, options, callback_) {
   }
 
   var req = new FSReqWrap();
-  req.oncomplete = callback;
+  req.oncomplete = makeCallback(callback);
```

Comment:
Does `makeCallback()` need to be called earlier? What if `callback` is invoked via `nullCheck()` a few lines up?